libfreerdp/core/nla.c

```diff
@@ -1134,13 +1140,104 @@ static BOOL nla_read_TSCspDataDetail(WinPrAsn1Decoder* dec, rdpSettings* setting
 	return set_creds_octetstring_to_settings(dec, 4, TRUE, FreeRDP_CspName, settings);
 }
 
+static BOOL nla_read_KERB_TICKET_LOGON(rdpNla* nla, wStream* s, KERB_TICKET_LOGON* ticket)
+{
+	/* mysterious extra 16 bytes before TGS/TGT content */
+	if (Stream_GetRemainingLength(s) < 16 + 16)
+		return FALSE;
+
+	Stream_Read_UINT32(s, ticket->MessageType);
+	Stream_Read_UINT32(s, ticket->Flags);
+	Stream_Read_UINT32(s, ticket->ServiceTicketLength);
+	Stream_Read_UINT32(s, ticket->TicketGrantingTicketLength);
+
+	if (ticket->MessageType != KerbTicketLogon)
+	{
+		WLog_ERR(TAG, "Not a KerbTicketLogon");
+		return FALSE;
+	}
+
+	if (Stream_GetRemainingLength(s) <
+	    ticket->ServiceTicketLength + ticket->TicketGrantingTicketLength)
```

Comment:
granted, overflow unlikely, but best force to `64bit` with `1ull * ticket->ServiceTicketLength`